File: src/test/java/org/jx1/TestData.java
package org.jx1;

import java.io.ByteArrayOutputStream;
import java.util.Random;

/** Deterministic sample inputs, shared by tests and the golden-vector generator. */
public final class TestData {

    private TestData() {}

    /** Repetitive text: 360 bytes. */
    public static byte[] text() {
        return "abracadabra hocus pocus abracadabra ".repeat(10).getBytes(java.nio.charset.StandardCharsets.US_ASCII);
    }

    /** A far back-reference at offset ~2700, beyond the quick-mode (ZX7) limit of 2176. */
    public static byte[] farMatch() {
        var out = new ByteArrayOutputStream();
        byte[] block = new byte[200];
        new Random(1).nextBytes(block);
        out.writeBytes(block);
        out.writeBytes("x".repeat(2500).getBytes(java.nio.charset.StandardCharsets.US_ASCII));
        out.writeBytes(block);
        return out.toByteArray();
    }

    /** Pseudo-random word soup: mixed literals and nearby matches. */
    public static byte[] wordSoup() {
        var random = new Random(42);
        byte[][] words = new byte[20][];
        for (int i = 0; i < words.length; i++) {
            words[i] = new byte[3 + random.nextInt(7)];
            random.nextBytes(words[i]);
        }
        var out = new ByteArrayOutputStream();
        for (int i = 0; i < 400; i++) {
            out.writeBytes(words[random.nextInt(words.length)]);
            out.write(' ');
        }
        return out.toByteArray();
    }
}

File: csharp/tests/Jx1.Tests/TestData.cs
using System.Text;

namespace Jx1.Tests;

/// <summary>Deterministic samples shared by the tests and Java golden vectors.</summary>
internal static class TestData
{
    internal static byte[] Text() =>
        Encoding.ASCII.GetBytes(string.Concat(Enumerable.Repeat(
            "abracadabra hocus pocus abracadabra ",
            10)));

    internal static byte[] FarMatch()
    {
        var block = new byte[200];
        new JavaRandom(1).NextBytes(block);

        using var output = new MemoryStream();
        output.Write(block);
        output.Write(Enumerable.Repeat((byte)'x', 2_500).ToArray());
        output.Write(block);
        return output.ToArray();
    }

    internal static byte[] WordSoup()
    {
        var random = new JavaRandom(42);
        var words = new byte[20][];
        for (int index = 0; index < words.Length; index++)
        {
            words[index] = new byte[3 + random.NextInt(7)];
            random.NextBytes(words[index]);
        }

        using var output = new MemoryStream();
        for (int index = 0; index < 400; index++)
        {
            output.Write(words[random.NextInt(words.Length)]);
            output.WriteByte((byte)' ');
        }
        return output.ToArray();
    }
}
